### PR TITLE
Api end points

### PR DESCRIPTION
**Pull request by boshr-tc**
EQM - API Endpoint - Post Questionnaire Template

### CODE REVIEW

File: EQM-GQE.API/Controllers/QuestionnaireController.cs [hunk 1]
```diff
@@ -46,5 +54,32 @@ public IEnumerable<Questionnaire> Get()
             })
             .ToArray();
         }
+
+        [HttpPost]
+        public async Task<ActionResult> CreateQuestionnaire(Questionnaire oQuestionnaire)
+        {
+            Questionnaire questionnaire = new ()
```
Comment (MOULAST) [inline EQM-GQE.API/Controllers/QuestionnaireController.cs]: We should be passing the oQuestionnaire into the logic layer. Then the logic layer will be responsible for calling the Data Layer and storing it in the database.

Comment (MOULAST) [reply EQM-GQE.API/Controllers/QuestionnaireController.cs]: Also, I would like to see some unit tests for both the logical layer and the API layer.

The api layer should test for error catching like a null questionnaire and other cases. Right now this endpoint will always respond with Ok which is not the case if there is an error.

File: EQM-GQE.API/Controllers/QuestionnaireController.cs [hunk 2]
```diff
@@ -27,11 +28,18 @@ public class QuestionnaireController : ControllerBase
             "{ 'pages': [ { 'name': 'page10', 'elements': [ { 'type': 'text', 'name': 'question1' }, { 'type': 'checkbox', 'name': 'question2', 'choices': [ 'item1', 'item2', 'item3' ] } ] } ] }" 
         };
 
-        private readonly ILogger<QuestionnaireController> _logger;
+        //private readonly ILogger<QuestionnaireController> _logger;
 
-        public QuestionnaireController(ILogger<QuestionnaireController> logger)
+        private readonly IQuestionnaireRepository _questionnaireRepository;
```
Comment (MOULAST) [inline EQM-GQE.API/Controllers/QuestionnaireController.cs]: you can probably move this to the logical layer once it is implemented and pass in the logical interface instead.

**Review by MOULAST — CHANGES_REQUESTED** [review]
looks like we have logical code in our front end API layer, let's move this to a logical layer and pass in an interface to interact with. The logic layer will work with the data layer. This will fee redundant for the post. but when we have more complex logical to implement we will be happy to have it.

Also I don't see any unit tests for this. I would like to see them for logical (testing business logic and mocking the data layer) and API (Testing endpoints for different error cases and happy path (again mocking the data layer).

**Comment by MOULAST** [comment]
@boshr-tc can we auto migrate the DB as well on start of the application to ensure we have the most up to date DB.

File: EQM-GQE.API/Controllers/QuestionnaireController.cs (hunk 2 above)
Comment (boshr-tc) [reply EQM-GQE.API/Controllers/QuestionnaireController.cs]: Moved private readonly IQuestionnaireRepository _questionnaireRepository; to the logical layer once it is implemented and pass in the logical interface instead.

File: EQM-GQE.API/Controllers/QuestionnaireController.cs (hunk 1 above)
Comment (boshr-tc) [reply EQM-GQE.API/Controllers/QuestionnaireController.cs]: Passed the oQuestionnaire into the logic layer. Then the logic layer will be responsible for calling the Data Layer and storing it in the database.

Confirmed with Stephen, we will close this PR without the Unit test work, Rajib will create a new branch and work on the unit test.

Comment (MOULAST) [reply EQM-GQE.API/Controllers/QuestionnaireController.cs]: Miscommunication, I want the unit test done as part of this PR.

**Review by MOULAST — APPROVED** [review]
looks good now

**Pull request merged**